src/receive_shipment/ReceiveShipmentDialog.jsx

```diff
@@ -13,22 +13,25 @@ const ReceiveShipmentDialog = ({
   viewOnly = false,
 }) => {
   const [filledForm, setFilledForm] = useState([]);
-
   const rowData = useMemo(() => {
-    const manifest = parts[0]?.manifest;
-    if (manifest)
-      return manifest.map((e) => {
-        return {
-          label: parts[0].label,
-          id: e.item._id,
-          batch: e.item.batch,
-          orderNumber: e.item.orderNumber,
-          partDescription: e.item.partDescription,
-          partNumber: e.item.partNumber,
-          partRev: e.item.partRev,
-          qty: e.qty,
-        };
-      });
+    if (parts?.length > 0) {
+      const manifest = parts[0]?.manifest;
+      if (manifest)
+        return manifest.map((e) => {
+          return {
+            label: parts[0]?.label,
+            id: e.item._id,
+            batch: e.item.batch,
+            orderNumber: e.item.orderNumber,
+            partDescription: e.item.partDescription,
+            partNumber: e.item.partNumber,
+            partRev: e.item.partRev,
+            qty: e.qty,
+            qtyReceived: 0,
```

Comment:
Its possible this might be non-zero if we take the same entry mutiple times. To be safe, it should probably be something like: `qtyReceived: e.qtyReceived || 0,`